merkletree/proof.go

```diff
@@ -80,38 +80,51 @@ func (ap *AuthenticationPath) authPathHash() []byte {
 	return hash
 }
 
-func (ap *AuthenticationPath) verifyBinding(key, value []byte) bool {
+// VerifyBinding verifies both the value and the commitment
+// of the proof node of the authentication path.
+//
+// This should be called only when ap is a proof of inclusion.
+func (ap *AuthenticationPath) VerifyBinding(key, value []byte) bool {
```

Comment:
> you saying that the ap.ProofType is set by the server? It is computed by the client when it receives an auth path.

Ah, my bad. I missed that somehow. I assumed `ap.ProofType()` was just a getter for the proof type, and that the server was setting the proof type on its end. Ignore my comment :)